### PR TITLE
refactor: rename new to empty

### PR DESCRIPTION
**Pull request by jaschdoc**
- `Collection.new` -> `Collection.empty`
- `Ref.new` -> `Ref.to`
- `Random.new` -> `Random.fresh`

Closes https://github.com/flix/flix/issues/7329

### CODE REVIEW

**Comment by jaschdoc** [comment]
Lots failures on the community build obviously

**Comment by magnus-madsen** [comment]
@jaschdoc A compiler test is also failing.

File: main/src/library/Array.flix [hunk 1]
```diff
@@ -408,7 +408,7 @@ mod Array {
     ///
     /// Returns the result of applying `f` to every element in `a`.
     ///
-    /// The result is a new array.
+    /// The result is an empty array.
```
Comment (magnus-madsen) [inline main/src/library/Array.flix:411]: @jaschdoc The search and replace here is problematic. In most cases its NOT an empty array. The documentation is trying to indicate that what is returned is new array (but it still contains elements).

Comment (magnus-madsen) [reply main/src/library/Array.flix:411]: I will go a head and merge this, but can you restore Array.flix and manually restore the necessary comments in a new PR?

**Review by magnus-madsen — APPROVED** [review]

File: main/src/library/Array.flix (hunk 1 above)
Comment (jaschdoc) [reply main/src/library/Array.flix:411]: > @jaschdoc The search and replace here is problematic. In most cases its NOT an empty array. The documentation is trying to indicate that what is returned is new array (but it still contains elements).

This was deliberate hehe oops

**Pull request merged**